### PR TITLE
Fixes & improvements.

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Various fixes & improvements.

### CODE REVIEW

File: MerlinAU.sh [hunk 1]
```diff
@@ -487,10 +487,9 @@ logo() {
 ##-----------------------------------------------##
 _CheckForNewScriptUpdates_()
 {
-
-   echo ""
    local DLRepoVersionNum  ScriptVersionNum
 
+   echo ""
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:492]: Approved

File: MerlinAU.sh [hunk 2]
```diff
@@ -1702,23 +1701,29 @@ get_required_space() {
...
+         fi
     done
 
-    for service_name in stop_conn_diag stop_samba stop_nasapps
+    for service_name in conn_diag samba nasapps
     do
         procNum="$(ps w | grep -w "$service_name" | grep -cv "grep -w")"
-        printf "$service_name: [$procNum]\n"
-        [ "$procNum" -gt 0 ] && service "$service_name" && sleep 1
+        if [ "$procNum" -gt 0 ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1722]: Approved

File: MerlinAU.sh [hunk 3]
```diff
@@ -3184,15 +3192,13 @@ Please manually update to version $minimum_supported_version or higher to use th
     ##------------------------------------------##
     Say "-----------------------------------------------------------"
     # List & log the contents of the ZIP file #
-    unzip -l "$FW_ZIP_FPATH" 2>&1 | while IFS= read -r uzLINE; do
-        Say "$uzLINE"
-    done
+    unzip -l "$FW_ZIP_FPATH" 2>&1 | \
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:3195]: Approved

File: MerlinAU.sh [hunk 4]
```diff
@@ -4118,14 +4124,15 @@ _ShowAdvancedOptionsMenu_()
...
+   then
        # Retrieve the current backup settings
        local current_backup_settings="$(Get_Custom_Setting "FW_Auto_Backupmon")"
 
-       printf "\n  ${GRNct}6${NOct}.  Toggle Auto-Backups"
+       printf "\n ${GRNct}ab${NOct}.  Toggle Automatic Backups"
        if [ "$current_backup_settings" = "DISABLED" ]
-       then printf "\n${padStr}[Current Build Type: ${REDct}${current_backup_settings}${NOct}]\n"
-       else printf "\n${padStr}[Current Build Type: ${GRNct}${current_backup_settings}${NOct}]\n"
+       then printf "\n${padStr}[Currently ${REDct}${current_backup_settings}${NOct}]\n"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4134]: Approved. Nice catch

File: MerlinAU.sh [hunk 5]
```diff
@@ -4198,17 +4206,20 @@ _advanced_options_menu_()
...
-            6) _Toggle_Auto_Backups_ && _WaitForEnterKey_
+            4) _toggle_change_log_check_
                ;;
-            7) if echo "$PRODUCT_ID" | grep -q "^GT-"
-               then change_build_type
-               else _InvalidMenuSelection_
-               fi
+            5) _toggle_beta_updates_
                ;;
+            ab) if [ -f "/jffs/scripts/backupmon.sh" ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4213]: approved.

File: MerlinAU.sh [hunk 6]
```diff
@@ -3079,10 +3086,11 @@ Please manually update to version $minimum_supported_version or higher to use th
         ## Modified by ExtremeFiretop [2024-Mar-20] ##
         ##------------------------------------------##
         # Check for the presence of backupmon.sh script
-        if [ -f "/jffs/scripts/backupmon.sh" ]; then
+        if [ -f "/jffs/scripts/backupmon.sh" ]
+        then
             local current_backup_settings="$(Get_Custom_Setting "FW_Auto_Backupmon")"
-			if [ "$FW_Auto_Backupmon" = "ENABLED" ]
-			then
+            if [ "$current_backup_settings" = "ENABLED" ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:3092]: Nice catch, approved.

**Review by ExtremeFiretop — APPROVED** [review]

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

How do we feel about pulling the Cron schedule back out from the advanced options back to the main menu?

image
image

Any preferences or thoughts?

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> How do we feel about pulling the Cron schedule back out from the advanced options back to the main menu?
> ...
> Any preferences or thoughts?
>
Looks good to me. The F/W Update check schedule & the postponement days are indeed related items so putting them back together in the same menu section is much better. Excellent idea!

**Pull request merged**